lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -287,13 +287,20 @@ protected function gatherRowData(array $data, array &$id, array &$nonemptyCompon
                     $dqlAlias = $cacheKeyInfo['dqlAlias'];
                     $type     = $cacheKeyInfo['type'];
 
+                    if(
+                        isset($cacheKeyInfo['discriminatorColumn']) && 
+                        isset($data[$cacheKeyInfo['discriminatorColumn']]) &&
+                        $data[$cacheKeyInfo['discriminatorColumn']] != $cacheKeyInfo['discriminatorValue']
```

Comment:
it seems to me that this change is breaking another test (/code/tests/Doctrine/Tests/ORM/Functional/SecondLevelCacheJoinTableInheritanceTest.php:179). Is it ok?